x-pack/plugins/file_upload/public/components/index_settings.js

```diff
@@ -0,0 +1,206 @@
+            defaultMessage="Index name"
+          />
+        }
+        isInvalid={indexNameError !== ''}
+        error={[indexNameError]}
+      >
+        <EuiFieldText
+          disabled={disabled}
+          placeholder={intl.formatMessage({
+            id: 'xpack.file_upload.indexNamePlaceholder',
+            defaultMessage: 'index name'
+          })}
+          value={indexName}
+          onChange={onIndexChange(setIndexName, setIndexNameError, indexNames)}
+          isInvalid={indexNameError !== ''}
+          aria-label={intl.formatMessage({
+            id: 'xpack.file_upload.indexNameAriaLabel',
+            defaultMessage: 'Index name, required field'
+          })}
+        />
+      </EuiFormRow>
+
+      <EuiSpacer size="s" />
+
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.file_upload.indexPatternLabel"
+            defaultMessage="Index pattern"
+          />
+        }
+        isInvalid={indexPatternError !== ''}
+        error={[indexPatternError]}
+      >
+        <EuiFieldText
+          disabled={disabled || !indexName}
+          placeholder={indexName}
+          value={indexPattern}
+          onChange={onIndexPatternChange(
+            setIndexPattern, setIndexPatternError, indexPatterns, indexName
+          )}
+          isInvalid={indexPatternError !== ''}
+        />
+      </EuiFormRow>
+    </Fragment>
+  );
+});
+
+function onIndexChange(setIndex, setIndexNameError, indexNames) {
+  return ({ target }) => {
+    const name = target.value;
+    const errorMessage = isIndexNameValid(name, indexNames);
+    setIndexNameError(errorMessage);
+    setIndex(name);
+  };
+}
+
+function onIndexPatternChange(
+  setIndexPattern, setIndexPatternError, indexPatterns, indexName
+) {
+  return ({ target }) => {
+    const pattern = target.value;
+    const errorMessage = isIndexPatternValid(pattern, indexPatterns, indexName);
+    setIndexPatternError(errorMessage);
+    setIndexPattern(pattern);
+  };
+}
+
+function isIndexNameValid(name, indexNames) {
+  if (indexNames.find(i => i === name)) {
+    return (
+      <FormattedMessage
+        id="xpack.file_upload.indexNameAlreadyExistsErrorMessage"
+        defaultMessage="Index name already exists"
+      />
+    );
+  }
+
+  const reg = new RegExp('[\\\\/\*\?\"\<\>\|\\s\,\#]+');
+  if (
+    (name !== name.toLowerCase()) || // name should be lowercase
+    (name === '.' || name === '..')   || // name can't be . or ..
+    name.match(/^[-_+]/) !== null  || // name can't start with these chars
+    name.match(reg) !== null // name can't contain these chars
+  ) {
+    return (
+      <FormattedMessage
+        id="xpack.file_upload.indexNameContainsIllegalCharactersErrorMessage"
+        defaultMessage="Index name contains illegal characters"
+      />
+    );
+  }
+  return '';
+}
+
+function isIndexPatternValid(name, indexPatterns, index) {
+  // if a blank name is entered, the index name will be used so avoid validation
```

Comment:
We should probably only allow valid index-pattern names. Ie.* index-pattern names that match the name of the index. Otherwise, the index-pattern is unusable.